File: validate_schemas.ts
import * as fs from 'fs';
import * as path from 'path';

const themeDir = 'phenomena-skincare';
const sectionsDir = path.join(themeDir, 'sections');

console.log("Validating section schemas via TypeScript/Node...");
fs.readdirSync(sectionsDir).forEach(filename => {
  if (filename.endsWith('.liquid')) {
    const filePath = path.join(sectionsDir, filename);
    const content = fs.readFileSync(filePath, 'utf-8');
    
    const schemaMatch = content.match(/{%\s*schema\s*%}(.*?){%\s*endschema\s*%}/s);
    if (schemaMatch) {
      const schemaJson = schemaMatch[1].trim();
      try {
        JSON.parse(schemaJson);
        console.log(`✅ ${filename}: Valid Schema`);
      } catch (err: any) {
        console.log(`❌ ${filename}: Invalid Schema! Error: ${err.message}`);
        console.log("--- Schema Content ---");
        console.log(schemaJson);
        console.log("----------------------");
      }
    } else {
      console.log(`⚠️ ${filename}: No schema block found.`);
    }
  }
});

File: validate_schemas.py
import os
import json
import re

theme_dir = 'phenomena-skincare'
sections_dir = os.path.join(theme_dir, 'sections')

print("Validating section schemas...")
for filename in os.listdir(sections_dir):
    if filename.endswith('.liquid'):
        file_path = os.path.join(sections_dir, filename)
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Extract schema block
        match = re.search(r'{%\s*schema\s*%}(.*?){%\s*endschema\s*%}', content, re.DOTALL)
        if match:
            schema_json = match.group(1).strip()
            try:
                json.loads(schema_json)
                print(f"✅ {filename}: Valid Schema")
            except Exception as e:
                print(f"❌ {filename}: Invalid Schema! Error: {e}")
                print("--- Schema Content ---")
                print(schema_json)
                print("----------------------")
        else:
            print(f"⚠️ {filename}: No schema block found.")
